src/Data/Entity/Login.php

```diff
@@ -0,0 +1,56 @@
+<?php
+namespace App\Data\Entity;
+
+use App\Data\Authenticator;
+use App\Data\Entity;
+
+class Login extends Entity
+{
+
+    protected $transformer = 'App\Data\Transformer\LoginTransformer';
```

Comment:
Something like this:
```PHP
class Entity
{
	public function toArray()
	{
		return [
			'test' => 'example',
			'foo' => 'bar',
		];
	}
}

abstract class Transformer
{

	public function __construct(Entity $entity)	{
		$this->entity = $entity;
	}
	
	public function getData()
	{
		return $this->transform($this->entity->toArray());
	}
	
	abstract protected function transform(array $data);
}

class JsonTransformer extends Transformer
{
	
	protected function transform(array $data)
	{
		return json_encode($data);
	}
}

$entity = new Entity;
$transformer = new JsonTransformer($entity);
$transformer->getData();
```

I don't completely love calling getData at the end, might be able to do something differently.